gulpfile.js

```diff
@@ -361,6 +361,96 @@ gulp.task("sonarqube:scan", async () => {
   }
 });
 
+/** CI ********************************************************************************************/
+
+// Replace each vss-extension.json version with the current build number
+gulp.task("ci:azure:hotfix-extensions-version", () => {
+  const buildNumber = process.env.BUILD_BUILDID;
+
+  if (!buildNumber) {
+    throw new Error("Missing build number");
+  }
+
+  const vssExtensions = globby.sync([
+    path.join(SOURCE_DIR, "extensions", "*", "vss-extension.json"),
+  ]);
+
+  return mergeStream(
+    vssExtensions.map((vssExtension) =>
+      gulp
+        .src(vssExtension)
+        .pipe(
+          gulpJsonEditor((json) => ({
+            ...json,
+            version: `${json.version}.${buildNumber}`,
+          })),
+        )
+        .pipe(gulp.dest(path.dirname(vssExtension))),
+    ),
+  );
+});
+
+gulp.task("ci:azure:hotfix-tasks-version", () => {
+  const buildNumber = process.env.BUILD_BUILDID;
+
+  if (!buildNumber) {
+    throw new Error("Missing build number");
+  }
+
+  const tasks = globby.sync(["src/extensions/*/tasks/*/v*/*.json"]);
+
+  return mergeStream(
+    tasks.map((task) =>
+      gulp
+        .src(task)
+        .pipe(
+          gulpJsonEditor((json) => ({
+            ...json,
+            version: {
+              ...json.version,
+              Patch: parseInt(buildNumber, 10),
+            },
+          })),
+        )
+        .pipe(gulp.dest(path.dirname(task))),
+    ),
+  );
+});
+
+gulp.task("ci:azure:get-extensions-version", (done) => {
+  if (!process.env.BUILD_SOURCESDIRECTORY) {
+    throw new Error("Missing BUILD_SOURCESDIRECTORY");
+  }
+
+  function run(extension, extensionPrefix) {
+    const vssExtension = path.join(
+      process.env.BUILD_SOURCESDIRECTORY,
+      "build",
+      "extensions",
+      extension,
+      "vss-extension.json",
+    );
+    const { version } = fs.readJsonSync(vssExtension);
+
+    const extensionManifest = path.join(
+      process.env.BUILD_SOURCESDIRECTORY,
+      "build",
+      "extensions",
+      extension,
+      "vss-extension.json",
+    );
+    console.log(`Fetched ext version ${version} for ${extension}`);
+    console.log(`##vso[task.setvariable variable=${extensionPrefix}_EXT_NAME]${extensionManifest}`);
+    console.log(
+      `##vso[task.setvariable variable=${extensionPrefix}_VERSION;isOutput=true]${version}`,
+    );
```

Comment:
this log syntax mirrors what we would see from the Azure task. Does it need to match or just for astetics? 

if not, I would say to keep it distinctly different. But if so I would want to know why it's needed :D

Reply:
This syntax is not aesthetic at all but it's needed because it's [the way to define variables in AzDO](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/set-variables-scripts?view=azure-devops&tabs=bash)

It's used [here](https://github.com/SonarSource/sonar-scanner-vsts/blob/master/azure-pipelines.yml#L159)